### PR TITLE
fix: Fix importing datasource with same name but different plugin issue

### LINKED ISSUE #12527: [Bug]: If duplicate named datasource with different plugin type is present in org, importing the application using JSON file does not import actual datasource
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

As we are not exporting the datasource config object we are now dependent on the datasource name if the same is available in the org where we want to import the datasource. This should have the plugin check, as there can be a datasource with different plugin but same datasource name, this breaks the action execution as now the imported actions.

### Steps To Reproduce

1. Create a postgres datasource named `dummy_datasource`
2. Attach an action for this datasource
3. Export the application
4. Create new org with mongo datasource named `dummy_datasource` 
5. Try to import the application in newly created org
6. Check the action failure as the datasource is not configured correctly

### Public Sample App

_No response_

### Version

Cloud: Release

### LINKED ISSUE #12536: [Bug]: Importing App into an already Imported/Forked Org orphans the queries & creates a new dummy dataSource
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://user-images.githubusercontent.com/91450662/161453333-0ea4eecd-5401-4abe-be45-40c6524a5cbb.mov

### Steps To Reproduce

1. Create an app with data source, eg used - Postgres & export the app
2. Create a new org & import the app json created above
3. Observe Reconnect model appears & data source connects fine when all values are filled
4. Observe app loads fine & queries are tagged to proper data source as source app
5. Now import the same app json again into same organisation
6. Observe - no Reconnect modal appears as this data source is already connected - as expected
7. **Observe - After page opens all queries are failing & queries are orphaned without the data source mapped**
8. **Also observe - another data source with same name is created** 
9. **Expected - In this scenario - the 2nd import onwards should connect & work with original data source as created from 1st import & not create duplicates & make the queries orphan**
10. As this import is continued - new dummy data sources keeps creating but app is not usable

### Public Sample App

_No response_

### Version

Cloud/Release v1.6.17

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

> Earlier we were not doing the plugin check while importing the application in the destination org, this resulted in action getting connected to datasource with same name but different plugin type, which resulted in action failure. 

### Steps To Reproduce

1. Create a postgres datasource named dummy_datasource
2. Attach an action for this datasource
3. Export the application
4. Create new org with mongo datasource named dummy_datasource
5. Try to import the application in newly created org
6. Check the action failure as the datasource is not configured correctly

Fixes #12527 
Fixes #12536

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> JUnit TC
> Manual testing

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ImportExportApplicationServiceCEImpl.java [hunk 1]
```diff
@@ -1743,16 +1746,8 @@ private String sanitizeDatasourceInActionDTO(ActionDTO actionDTO,
...
-                    final DatasourceConfiguration dsAuthConfig = ds.getDatasourceConfiguration();
-                    if (dsAuthConfig != null && dsAuthConfig.getAuthentication() != null) {
-                        dsAuthConfig.getAuthentication().setAuthenticationResponse(null);
-                        dsAuthConfig.getAuthentication().setAuthenticationType(null);
-                    }
-                    return ds;
-                })
                 // For git import exclude datasource configuration
-                .filter(ds -> applicationId != null ? ds.getName().equals(datasource.getName()) : ds.softEquals(datasource))
+                .filter(ds -> ds.getName().equals(datasource.getName()) && datasource.getPluginId().equals(ds.getPluginId()))
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ImportExportApplicationServiceCEImpl.java:1750]: Earlier we were not doing the plugin check which resulted in action getting connected to datasource with same name but different plugin type, which resulted in action failure

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java [hunk 2]
```diff
@@ -1702,7 +1702,7 @@ public void checkoutRemoteBranch_presentInLocal_throwError() {
         StepVerifier
                 .create(applicationMono)
                 .expectErrorMatches(throwable -> throwable instanceof AppsmithException
-                        && throwable.getMessage().equals(AppsmithError.GIT_ACTION_FAILED.getMessage("checkout", "origin/branchInLocal already exists in remote")))
+                        && throwable.getMessage().equals(AppsmithError.GIT_ACTION_FAILED.getMessage("checkout", "origin/branchInLocal already exists in local - branchInLocal")))
```
Comment (abhvsn) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java:1705]: Update to fix testcase, though this is not related to the change we are doing in this PR

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java [hunk 3]
```diff
@@ -2497,6 +2497,7 @@ public void importApplicationFromGit_validRequestWithDuplicateDatasourceOfSameTy
         GitAuth gitAuth = gitService.generateSSHKey().block();
 
         ApplicationJson applicationJson =  createAppJson(filePath).block();
+        applicationJson.getExportedApplication().setName(null);
```
Comment (abhvsn) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java:2500]: Fixing the testcase which was failing earlier not related to this PR

**Review by AnaghHegde — DISMISSED** [review]

**Review by AnaghHegde — APPROVED** [review]

File: app/server/appsmith-server/src/test/java/com/appsmith/server/solutions/ImportExportApplicationServiceTests.java [hunk 4]
```diff
@@ -2235,5 +2235,58 @@ public void importApplication_datasourceWithSameNameAndDifferentPlugin_importedW
...
+
+        ApplicationJson applicationJson = createAppJson("test_assets/ImportExportServiceTest/valid-application.json").block();
+
+        Organization testOrganization = new Organization();
+        testOrganization.setName("Duplicate datasource with same plugin org");
+        testOrganization = organizationService.create(testOrganization).block();
+
+        Datasource testDatasource = new Datasource();
+        // Chose any plugin except for mongo, as json static file has mongo plugin for datasource
+        Plugin postgreSQLPlugin = pluginRepository.findByName("MongoDB").block();
```
Comment (abhvsn) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/solutions/ImportExportApplicationServiceTests.java:2251]: @AnaghHegde can you please change the name here and also update the comment

**Review by AnaghHegde — APPROVED** [review]

**Pull request merged**